test/Microsoft.Azure.Devices.Edge.Test/EdgeAgentDirectMethods.cs

```diff
@@ -61,6 +61,96 @@ public async Task TestGetModuleLogs()
             Assert.AreEqual(expected, response.Payload.Replace("\r\n", "\n"));
         }
 
+        [Test]
+        public async Task TestGetModuleLogsSinceDuration()
```

Comment:
Is it possible to write unit tests instead of end-to-end tests? This is a very high-level way to test low-level functionality, and will add as much as 15 minutes to the end-to-end run time.

For example, could tests be added to `edge-agent\test\Microsoft.Azure.Devices.Edge.Agent.Core.Test\requests\RequestHandlerBaseTest.cs`?

Reply:
It's not possible to use unit tests on this case, because what's causing the problem is the API call with invalid DateTime formats. If we only add tests to check the DateTime format after the serialization we can't reproduce the problem without calling the API.